src/main/java/net/fabricmc/mappingio/MappingReader.java

```diff
@@ -43,6 +43,26 @@ public static MappingFormat detectFormat(Path file) throws IOException {
 		if (Files.isDirectory(file)) {
 			return MappingFormat.ENIGMA_DIR;
 		} else {
+			String fileName = file.getFileName().toString();
+			int dotIdx = fileName.lastIndexOf('.');
+
+			if (dotIdx >= 0) {
+				String ext = fileName.substring(dotIdx + 1);
+
+				switch (ext) {
+				case "mapping": // Standard name
+				case "mappings": // Sometimes used for standalone (non-dir) files
```

Comment:
Im not convinved this is a good idea for more general names like mapping.

Reply:
Solved by adding the `trustFileExtension` parameter